packages/grammar-selector/lib/grammar-list-view.js

```diff
@@ -82,32 +82,64 @@ class GrammarListView {
   async toggle () {
     if (this.panel != null) {
       this.cancel()
-    } else if (atom.workspace.getActiveTextEditor()) {
-      this.editor = atom.workspace.getActiveTextEditor()
+      return
+    }
+
+    const editor = atom.workspace.getActiveTextEditor()
```

Comment:
It's trivial, but the original felt wasteful compared to just keeping the editor the first time. It's also technically more correct perhaps? Can the editor change between two consecutive steps?